source/getting-started/control-with-amdc/encoder-fb/index.md

```diff
@@ -1,13 +1,97 @@
+
+    // User to set encoder count per revolution
+    ENCODER_COUNT_PER_REV = 1024;
+
+    // Set 1 if encoder count increases with CCW rotation of shaft, Set 0 if encoder count increases with CW rotation of shaft
+    CCW = 1; 
+
+    // Angular position to be computed
+    double theta_m_enc;
+
+    // User to set encoder offset
+    enc_theta_m_offset = 100;
+
+
+    // Convert to radians
+    if (CCW){
+        theta_m_enc = (double) PI2 * ( ( (double)position - enc_theta_m_offset )/ (double) ENCODER_COUNT_PER_REV);
+    }
+    else{
+        theta_m_enc = (double) PI2 * ( ( (double)ENCODER_COUNT_PER_REV - (double)1 -(double)position + enc_theta_m_offset )/ (double) ENCODER_COUNT_PER_REV);
+    }
+    
+     // Mod by 2 pi
+    theta_m_enc = fmod(theta_m_enc,PI2 );
+    return theta_m_enc;
+}
+```
+
+
+
+### Finding the offset
+
+The example code shown above makes uses of an encoder offset value. It is necessary for the user to find this offset experimentally for their machine. For synchronous machines, this offset is the count value measured by the encoder when the d-axis of the rotor is aligned with the phase U winding axis of the stator. 
+
+To get the rotor to align with the phase U winding, the user may have some current flow through phase U and out of phase V and phase W. This can done by using a DC supply, with phase U connected to the positive terminal and phases V and W connected to the negative terminal. Alternately, the user may also inject some current along the d-axis using the AMDC [Signal Injection](https://docs.amdc.dev/getting-started/user-guide/injection/index.html) module. While injecting d-axis current, the user must ensure that the rotor position is set to zero in the control code. After obtaining the offset, the user needs to set the variable `enc_theta_m_offset` to the appropriate value in the `task_get_theta_m()` function. 
+
+To ensure that the obtained encoder offset is correct, the user may perform additional validation. For a permanent magnet synchronous motor, this can be done as follows:
+
+1. Spin the motor up-to a steady speed under no load conditions
+1. Measure the d-axis voltage commanded by the current regulator
+1. Repeat the experiment for a few different motor speeds
+1. Plot the d-axis voltage against the motor speed
+1. The d-axis voltage if the offset is tuned correctly, should be close to zero for all speeds.
+1. In-case there is an error in the offset value, a significant speed dependent voltage will appear on the d-axis voltage
+
+
 
 ## Computing Speed from Position
 
-- LPF
-- State Filter
-- Observer
\ No newline at end of file
+The user needs to compute a rotor speed signal from the obtained position signal to be used in the control algorithm. There are several ways to do this. A simple and straightforward way to do this would be to compute the discrete time derivative of the position signal in the controller as shown below. This can be referred to as $\Omega_{raw}$.
```

Comment:
```suggestion
The user needs to compute a rotor speed signal from the obtained position signal to be used in the control algorithm. There are several ways to do this. 

### Difference Equation Approach

A simple, but naive, way to do this would be to compute the discrete time derivative of the position signal in the controller as shown below. This can be referred to as $\Omega_{raw}$.
```